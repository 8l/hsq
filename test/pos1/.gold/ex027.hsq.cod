	top:top top sqmain

.  _a:0 _k:0 _n:55

_main:
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; bp 0
	bp; sp bp
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t1 0
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t2 0

	t1; t2; c1 t1; dec t1; t1 t2
	_k; t2 Z; Z _k; Z

	t1; t2; ?+11; _k Z; Z ?+4; Z; 0 t1; t1 t2
	t2 (-1)

	_k Z; ?+9; Z ?+5; Z; inc 0
	t1; t2; ?+11; _k Z; Z ?+4; Z; 0 t1; t1 t2
	_a; t2 Z; Z _a; Z

	_a (-1)

	_n (-1)

	_k Z; ?+9; Z ?+5; Z; dec 0
	t1; t2; ?+11; _k Z; Z ?+4; Z; 0 t1; t1 t2
	_a; t2 Z; Z _a; Z

	_a (-1)

	_n (-1)

	inc _n
	inc _n

	_n (-1)

	dec _n
	dec _n
	dec _n
	dec _n

	_n (-1)

	t1; t2; ?+11; _k Z; Z ?+4; Z; 0 t1; t1 t2
	_k Z; ?+9; Z ?+5; Z; inc 0
	_a; t2 Z; Z _a; Z

	_a (-1)

	_n (-1)

	t1; t2; ?+11; _k Z; Z ?+4; Z; 0 t1; t1 t2
	_k Z; ?+9; Z ?+5; Z; dec 0
	_a; t2 Z; Z _a; Z

	_a (-1)

	_n (-1)

	_a; _k Z; Z _a; Z

	t1; t2; ?+11; _a Z; Z ?+4; Z; 0 t1; t1 t2
	t2 (-1)

	?+23; ?+21; ?+24; _k Z; Z ?+10; Z ?+8
	Z ?+11; Z; 0; c2 Z; Z 0; Z

	_n (-1)

	t1; _n Z; Z t1; Z; inc _n
	t1 (-1)

	_n (-1)

	t1; _n Z; Z t1; Z; dec _n
	t1 (-1)

	_n (-1)

	t1; t2; ?+11; _k Z; Z ?+4; Z; 0 t1; t1 t2
	t2 (-1)

	?+8; sp ?+4; t2; 0 t2; inc sp
	?+8; sp ?+4; t1; 0 t1; inc sp
	sp; bp sp
	?+8; sp ?+4; bp; 0 bp; inc sp
	?+8; sp ?+4; ?+7; 0 ?+3; Z Z 0

sqmain:
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; ?+2 0 _main; . ?; inc sp

	Z Z (-1)

. c2:49 c1:_k

. t1:0 t2:0

. inc:-1 Z:0 dec:1 ax:0 bp:0 sp:-sp
