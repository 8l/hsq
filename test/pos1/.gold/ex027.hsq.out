788779556655112211